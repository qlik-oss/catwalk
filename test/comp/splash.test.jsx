import React from 'react';
import renderer from 'react-test-renderer';
import AppList from '../../src/components/app-list';
import ErrorInfo from '../../src/components/error-info';
import Splash from '../../src/components/splash';

const appList = <ul>list</ul>;
const errorInfo = <span>Some error</span>;
jest.mock('../../src/components/app-list', () => jest.fn());
AppList.mockImplementation(() => appList);
jest.mock('../../src/components/error-info', () => jest.fn());
ErrorInfo.mockImplementation(() => errorInfo);

describe('<Splash />', () => {
  it('should render <AppList />', () => {
    const error = null;
    const engineURL = '';
    const fetchDockList = true;

    const instance = renderer.create(<Splash
      error={error}
      engineURL={engineURL}
      fetchDocList={fetchDockList}
    />).toJSON();

    expect(instance).toMatchSnapshot();
  });

  it('should render <ErrorInfo /> when having error', () => {
    const error = new Error('An error occurred');
    const engineURL = '';

    const instance = renderer.create(<Splash
      error={error}
      engineURL={engineURL}
    />).toJSON();

    expect(instance).toMatchSnapshot();
  });

  it('should render <AppList /> even though error is provided', () => {
    const error = new Error('An error occurred');
    const engineURL = '';
    const fetchDockList = true;

    const instance = renderer.create(<Splash
      error={error}
      engineURL={engineURL}
      fetchDocList={fetchDockList}
    />).toJSON();

    expect(instance).toMatchSnapshot();
  });

  // it('should display message if empty docList array', () => {
  //   const docs = [];
  //   const error = null;
  //   const engineURL = '';

  //   const tree = renderer.create(<Splash
  //     docs={docs}
  //     error={error}
  //     engineURL={engineURL}
  //   />).toJSON();

  //   expect(tree).toMatchSnapshot();
  // });
});
