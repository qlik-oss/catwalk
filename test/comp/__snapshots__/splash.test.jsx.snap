// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Splash /> should render <AppList /> 1`] = `
Array [
  <div
    className="center-content"
  >
    <div
      className="splash"
    >
      <span
        className="SVGInline logo"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg class=\\"SVGInline-svg logo-svg\\" data-testid=\\"svg-element\\" />",
          }
        }
      />
      <ul>
        list
      </ul>
    </div>
  </div>,
  <div
    className="cat-with-bubble"
    style={null}
  >
    <span
      className="SVGInline close-me"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg class=\\"SVGInline-svg close-me-svg\\" data-testid=\\"svg-element\\" />",
        }
      }
      onClick={[Function]}
      title="Hide cat"
    />
    <span
      className="SVGInline cat"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg class=\\"SVGInline-svg cat-svg\\" data-testid=\\"svg-element\\" />",
        }
      }
    />
    <p
      className="bubble"
      onClick={[Function]}
      style={
        Object {
          "width": "7em",
        }
      }
    >
      Click my speech bubble to open the demo app.
    </p>
  </div>,
]
`;

exports[`<Splash /> should render <AppList /> even though error is provided 1`] = `
Array [
  <div
    className="center-content"
  >
    <div
      className="splash"
    >
      <span
        className="SVGInline logo"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg class=\\"SVGInline-svg logo-svg\\" data-testid=\\"svg-element\\" />",
          }
        }
      />
      <ul>
        list
      </ul>
    </div>
  </div>,
  <div
    className="cat-with-bubble"
    style={null}
  >
    <span
      className="SVGInline close-me"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg class=\\"SVGInline-svg close-me-svg\\" data-testid=\\"svg-element\\" />",
        }
      }
      onClick={[Function]}
      title="Hide cat"
    />
    <span
      className="SVGInline cat"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg class=\\"SVGInline-svg cat-svg\\" data-testid=\\"svg-element\\" />",
        }
      }
    />
    <p
      className="bubble"
      onClick={[Function]}
      style={
        Object {
          "width": "7em",
        }
      }
    >
      Click my speech bubble to open the demo app.
    </p>
  </div>,
]
`;

exports[`<Splash /> should render <ErrorInfo /> when having error 1`] = `
Array [
  <div
    className="center-content"
  >
    <div
      className="splash"
    >
      <span
        className="SVGInline error-logo"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg class=\\"SVGInline-svg error-logo-svg\\" data-testid=\\"svg-element\\" />",
          }
        }
      />
      <span>
        Some error
      </span>
    </div>
  </div>,
  <div
    className="cat-with-bubble"
    style={null}
  >
    <span
      className="SVGInline close-me"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg class=\\"SVGInline-svg close-me-svg\\" data-testid=\\"svg-element\\" />",
        }
      }
      onClick={[Function]}
      title="Hide cat"
    />
    <span
      className="SVGInline cat"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg class=\\"SVGInline-svg cat-svg\\" data-testid=\\"svg-element\\" />",
        }
      }
    />
    <p
      className="bubble"
      onClick={[Function]}
      style={
        Object {
          "width": "7em",
        }
      }
    >
      Click my speech bubble to open the demo app.
    </p>
  </div>,
]
`;
